pkg/image/containerdregistry/registry.go

```diff
@@ -261,3 +265,32 @@ func dropXattrs(h *tar.Header) (bool, error) {
 
 	return true, nil
 }
+
+// only unpack specified files (if not root)
+func fromFilter(from string) archive.Filter {
+	if from == image.Root {
+		// no-op, unpacks everything
+		return func(h *tar.Header) (bool, error) {
+			return true, nil
+		}
+	}
+
+	return func(h *tar.Header) (bool, error) {
+		// check if from is a dir or a file
+		if !h.FileInfo().IsDir() {
+			// we have a file
+			// only unpack this file
+			if from != h.FileInfo().Name() {
+				return false, nil
+			}
+			return true, nil
+		}
+		// we have a directory
+		// unpack the directory and all files under this directory
+		// TODO ensure directory is an exact match and only files under the heirarchy are pulled
+		if from == h.FileInfo().Name() || strings.Contains(from, h.FileInfo().Name()) {
```

Comment:
I don't think this will pull the hierarchy correctly

https://play.golang.org/p/MB2vgqucZhG

If I'm correct, then I think the simplest fix would be to use `strings.HasPrefix` instead: https://pkg.go.dev/strings#HasPrefix

/hold